src/launch/types.rs

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::certs::sev;
+use crate::launch::{Header, Measurement, Policy, Session};
+
+use std::marker::PhantomData;
+
+/// Initialize the SEV platform context.
+#[repr(C)]
+pub struct Init;
+
+/// Initiate SEV launch flow.
+#[repr(C)]
+pub struct LaunchStart<'a> {
+    pub handle: u32,
+    policy: Policy,
+    dh_addr: u64,
+    dh_len: u32,
+    session_addr: u64,
+    session_len: u32,
+    _phantom: PhantomData<&'a ()>,
+}
+
+impl<'a> LaunchStart<'a> {
+    pub fn new(policy: &'a Policy, dh: &'a sev::Certificate, session: &'a Session) -> Self {
+        Self {
+            handle: 0, /* platform will generate one for us */
+            policy: *policy,
+            dh_addr: dh as *const _ as _,
+            dh_len: std::mem::size_of_val(dh) as _,
```

Comment:
Remember `size_of_val()` is computed at runtime, while `size_of()` at compilation time. I would use `size_of()` wherever possible.
```suggestion
            dh_len: std::mem::size_of::<sev::Certificate>() as _,
```

Reply:
I think `size_of_val()` is only a runtime overhead if I pass in a slice or a trait object.

I initially reached for `size_of_val()` because I felt it more "visually" coupled the correct size value with the variable I'm using, given the bugs I've had to solve from the creation of these FFI structs I felt that was helpful as an author.

Reply:
ah, you are right